wasm/src/main/scala/org/scalajs/linker/backend/wasmemitter/VarGen.scala

```diff
@@ -484,7 +500,11 @@ object VarGen {
 
     def forFunction(idx: Int): WasmTypeName = WasmTypeName(s"f.$idx")
 
-    def forRecFunction(idx: Int): WasmTypeName = WasmTypeName(s"recf.$idx")
+    val cloneFunctionType = WasmTypeName("cloneFuncType")
+    val isJSClassInstanceFuncType = WasmTypeName("isJSClassInstanceFuncType")
+
+    def forTableFunctionType(methodName: MethodName): WasmTypeName =
+      WasmTypeName("m." + methodName.nameString)
```

Comment:
Now, the name starts with `m.m_...` like `$m.m_Ljava.lang.String`. I personally don't mind, but maybe we don't need `m.` prefix anymore? 😄

Reply:
It's not great but I believe keeping the `m.` is more consistent with all the other things. This way we're sure that we everything is well namespaced.

Eventually I'd rather get rid of the `m_` part instead of the `m.` part, but that will require more important changes not to rely on `nameString`.

Or maybe we could do `.nameString.substring(2)` to get rid of the `m_`? Would that be better in your opinion?

Reply:
I'll merge it as is for now, to unblock my stack of PRs. We can still easily tweak this encoding later.